src/main/java/seedu/duke/helper/Parser.java

```diff
@@ -19,4 +22,71 @@
     public static String[] commandParser(String userInput) {
         return userInput.trim().split("/info");
     }
+
+    public static String[] parseAddPatient(String parameters) {
+        // Currently the basic input order should be to just follow the order
+        // in Patient class: nric, fullName, age, gender, address, dob, dateAdmission
+        String[] parametersArray = parameters.split(",");
+        for (String parameter: parametersArray) {
+            parameter = parameter.trim();
+        }
+        
+
+        return null;
+    }
+
+    private static boolean validateNric(String nric) {
+        Pattern nricPattern = Pattern.compile("[A-Z][0-9]{7}[A-Z]");
+        Matcher nricMatcher = nricPattern.matcher(nric);
+        return nricMatcher.matches();
+    }
+
+    private static boolean validateFullName(String fullName) {
+        Pattern fullNamePattern = Pattern.compile("([a-zA-Z]+( [a-zA-Z]+)+)");
+        Matcher fullNameMatcher = fullNamePattern.matcher(fullName);
+        return fullNameMatcher.matches();
+    }
+
+    private static boolean validateAge(String ageString) {
+        int age = Integer.parseInt(ageString);
+        if (1 <= age && age <= 120) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private static boolean validateGender(String gender) {
```

Comment:
am thinking perhaps can abstract out all the validators to a validator class separate from parser

Reply:
Also can. But I think let's focus on getting all the three Patient, Medicine and Doctor up first before thinking about more OOP refactoring